models/ROMS_rutgers/model_mod.f90

```diff
@@ -0,0 +1,1716 @@
+
+character(len=*), parameter   :: routine = 'parse_variable_input'
+character(len=*), intent(in)  :: state_variables(:)
+integer,          intent(out) :: ngood
+
+integer :: i
+character(len=NF90_MAX_NAME) :: varname       ! column 1
+character(len=NF90_MAX_NAME) :: dartstr       ! column 2
+character(len=NF90_MAX_NAME) :: minvalstring  ! column 3
+character(len=NF90_MAX_NAME) :: maxvalstring  ! column 4
+character(len=NF90_MAX_NAME) :: state_or_aux  ! column 5   change to updateable
+
+ngood = 0
+MyLoop : do i = 1, MAX_STATE_VARIABLES
+
+   varname      = trim(state_variables(table_columns*i-4))
+   dartstr      = trim(state_variables(table_columns*i-3))
+   minvalstring = trim(state_variables(table_columns*i-2))
+   maxvalstring = trim(state_variables(table_columns*i-1))
+   state_or_aux = trim(state_variables(table_columns*i  ))
+
+   if (varname == ' ' .and. dartstr == ' ') exit MyLoop ! Found end of list.
+
+   if (varname == ' ' .or. dartstr == ' ') then
+      string1 = 'model_nml:model "variables" not fully specified'
+      call error_handler(E_ERR, routine, string1, source)
+   endif
+
+   ! Make sure DART kind is valid
+   if(get_index_for_quantity(dartstr) < 0) then
+      write(string1, '(''there is no quantity <'',a,''> in obs_kind_mod.f90'')') trim(dartstr)
+      call error_handler(E_ERR, routine, string1, source)
+   endif
+
+   call to_upper(minvalstring)
+   call to_upper(maxvalstring)
+   call to_upper(state_or_aux)
+
+   var_names(   i) = varname
+   kind_list(   i) = get_index_for_quantity(dartstr)
+   clamp_vals(i,1) = string_to_real(minvalstring)
+   clamp_vals(i,2) = string_to_real(maxvalstring)
+   update_list( i) = string_to_logical(state_or_aux, 'UPDATE')
+
+   ngood = ngood + 1
+
+enddo MyLoop
+
+if (ngood == MAX_STATE_VARIABLES) then
+   string1 = 'WARNING: There is a possibility you need to increase ''MAX_STATE_VARIABLES'''
+   write(string2, '(''WARNING: you have specified at least '',i4,'' perhaps more.'')') ngood
+   call error_handler(E_MSG, routine, string1, source, text2=string2)
+endif
+
+end subroutine parse_variable_input
+
+
+!-----------------------------------------------------------------------
+! Find the named variable (often 'ocean_time') in a ROMS netCDF file.
+! If it is not found, it is a fatal error.
+!
+! filename:        Name of ROMS file (used to generate useful error messages)
+! ncid:            NetCDF handle to the ROMS netCDF file
+! var_name:        Name which contains the time
+! dim_name:        Dimension name of the time variable
+! myvarid:         Optional output of the variable id 
+! calendar:        Character string indicating the calendar in use
+! last_time_index: Value of the last time dimension
+! last_time:       Time/date of the last time
+! origin_time:     Base time other times are relative to
+! all_times:       Array of all times in the variable
+
+subroutine get_time_information(filename, ncid, var_name, dim_name, myvarid, &
```

Comment:
I think this routine is from the other ROMS (ucla) for creating roms obs. I'd pull it out of this model if it is not being used for ROMS-rutgers.

https://github.com/NCAR/DART/blob/f7205cb8bb947d2eb3580e5e81e77e5720430868/observations/obs_converters/ROMS/convert_roms_obs.f90#L56

Reply:
sounds good